fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/BestMvSelector.java

```diff
@@ -50,10 +57,19 @@ public OptExpression selectBest() {
         for (OptExpression expression : expressions) {
             calculateStatistics(expression, context);
         }
+
+        // collect original table scans
+        List<Table> originalTables = MvUtils.getAllTables(queryPlan);
+        Set<ScalarOperator> predicates = MvUtils.getAllValidPredicates(queryPlan);
+        Set<String> equivalenceColumns = Sets.newHashSet();
+        Set<String> nonEquivalenceColumns = Sets.newHashSet();
+        MvUtils.splitPredicate(predicates, equivalenceColumns, nonEquivalenceColumns);
```

Comment:
`predicates ` seems a bit wasting only for sort keys or other index keys?  maybe we can only collect predicates for scan operator?

Reply:
ok, I will optimize it.